planner/core/planbuilder.go

```diff
@@ -1358,6 +1359,47 @@ func (b *planBuilder) buildDDL(node ast.DDLNode) Plan {
 				table:     v.ReferTable.Name.L,
 			})
 		}
+		if v.Select != nil {
```

Comment:
Can we wrap these code into a `buildCreateTable` function to make the code clearer?